Source: Upstream
Reason: Fix build with musl

From d2bb019533bbcbd1cee3891822ac7ba6ff55ad3d Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Fri, 26 Jul 2013 12:39:51 +0100
Subject: [PATCH] configure: Use AC_USE_SYSTEM_EXTENSIONS

...instead of detecting glibc and using _GNU_SOURCE manually.  This
should fix the build when using glibc-emulating libraries; we can
defer portability work to autoconf.

https://bugzilla.gnome.org/show_bug.cgi?id=684123
---
 configure.ac | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index d309f28..9443709 100644
--- a/configure.ac
+++ b/configure.ac
@@ -111,6 +111,7 @@ AC_SUBST(LT_CURRENT_MINUS_AGE)
 dnl Checks for programs.
 AC_PROG_CC
 AC_PROG_CPP
+AC_USE_SYSTEM_EXTENSIONS
 
 AM_CONDITIONAL(HAVE_GCC, [test "$GCC" = "yes"])
 
@@ -231,11 +232,6 @@ if test "x$glib_have_cocoa" = "xyes"; then
   LDFLAGS="$LDFLAGS -Wl,-framework,Foundation"
 fi
 
-gl_GLIBC21
-AS_IF([test "x$GLIBC21" = "xyes"], [
-  AC_DEFINE([_GNU_SOURCE], 1, [Make all glibc extensions visible])
-])
-
 dnl declare --enable-* args and collect ac_help strings
 AC_ARG_ENABLE(debug,
               AS_HELP_STRING([--enable-debug=@<:@no/minimum/yes@:>@],
-- 
1.8.4

